pkg/controller/provider/validation.go

```diff
@@ -400,3 +412,429 @@ func isValidNFSPath(nfsPath string) bool {
+	r.Log.V(3).Info("SSH validation: key details for testing",
+		"provider", provider.Name,
+		"publicKeyLength", len(publicKey),
+		"publicKeyPrefix", func() string {
+			if len(publicKey) > 40 {
+				return publicKey[:40] + "..."
+			}
+			return publicKey
+		}(),
+		"restrictedKeyLength", len(restrictedKey),
+		"privateKeyLength", len(privateKeyBytes))
+
+	// Test SSH connectivity on ALL hosts (don't stop early)
+	r.Log.Info("SSH validation: testing all hosts for complete status",
+		"provider", provider.Name,
+		"totalHosts", len(hostsToTest))
+	failedHosts := []string{}
+	successHosts := []string{}
+
+	// Test all hosts to provide complete status
+	for i := range hostsToTest {
+		host := &hostsToTest[i]
+		if host.ip == "" {
+			r.Log.Info("SSH validation: host has no management IP - marking as failed",
+				"provider", provider.Name,
+				"hostName", host.name)
+			failedHosts = append(failedHosts, fmt.Sprintf("%s (no management IP)", host.name))
+			continue
+		}
+		r.Log.V(3).Info("SSH validation: testing host",
+			"provider", provider.Name,
+			"hostName", host.name,
+			"hostIP", host.ip,
+			"hostIndex", i+1,
+			"totalHosts", len(hostsToTest))
+		hostResult := r.testSSHConnectivity(host.ip, privateKeyBytes)
+		r.Log.V(3).Info("SSH validation: host test result",
+			"provider", provider.Name,
+			"hostName", host.name,
+			"hostIP", host.ip,
+			"success", hostResult)
+
+		if hostResult {
+			successHosts = append(successHosts, fmt.Sprintf("%s (%s)", host.name, host.ip))
+		} else {
+			failedHosts = append(failedHosts, fmt.Sprintf("%s (%s)", host.name, host.ip))
+		}
+	}
+
+	r.Log.Info("SSH validation: all hosts tested",
+		"provider", provider.Name,
+		"successCount", len(successHosts),
+		"failedCount", len(failedHosts))
+
+	// If all hosts passed, we're done
+	if len(failedHosts) == 0 {
+		r.Log.Info("SSH validation: all hosts have SSH working",
+			"provider", provider.Name,
+			"hostCount", len(hostsToTest))
+		provider.Status.DeleteCondition(SSHReadiness)
+		return nil
+	}
+
+	// Build detailed message with both success and failure lists
+	var message strings.Builder
+	if len(successHosts) == 0 {
+		message.WriteString(fmt.Sprintf("SSH connectivity failed for all %d ESXi host(s). Manual SSH key installation required.\n\n", len(hostsToTest)))
+	} else {
+		message.WriteString(fmt.Sprintf("SSH connectivity: %d succeeded, %d failed (of %d total host(s)).\n\n",
+			len(successHosts), len(failedHosts), len(hostsToTest)))
+	}
+
+	if len(successHosts) > 0 {
+		message.WriteString(fmt.Sprintf("✓ SSH WORKING (%d hosts):\n", len(successHosts)))
+		for _, host := range successHosts {
+			message.WriteString(fmt.Sprintf("  • %s\n", host))
+		}
+		message.WriteString("\n")
+	}
+
+	if len(failedHosts) > 0 {
+		message.WriteString(fmt.Sprintf("✗ SSH SETUP NEEDED (%d hosts):\n", len(failedHosts)))
+		for _, host := range failedHosts {
+			message.WriteString(fmt.Sprintf("  • %s\n", host))
+		}
+		message.WriteString("\n")
+	}
```

Comment:
You specify the hosts in the condition items do we need to have it in the message aswell? This applies to the passed hosts, do we need to mention the working hosts? It goes against how we build the condition for other validations.

Reply:
the condition is made of 3 main things
1. type -> if it's error 
2. message -> what's the problems, should be short describtion.
3. items -> List of items to which this applies to, mainly used for VMs but can be anything. UI team would need to finish the visualization of that for any other item list.

So if we will want to fit everything inside the message that will go agaist the design.